osbuild/buildroot.py

```diff
@@ -237,6 +237,12 @@ def run(self, argv, monitor, timeout=None, binds=None, readonly_binds=None, extr
         mounts += ["--ro-bind-try",
                    os.path.join(self._rootdir, "etc/containers"),
                    "/etc/containers"]
+        mounts += ["--ro-bind-try",
+                   os.path.join(self._rootdir, "ostree"),
```

Comment:
@ondrejbudai This is a bit of a wrinkle AIUI - we need to mount the "/ostree" from the container that we are running from into the buildroot because `bootc install to-filesystem --source-imgref=...` will otherwise fail with `ERROR Gathering source info from an external source: Task Reading ostree commit failed:`. Maybe I'm missing something but shouldn't the `--source-imgref` be used and mounted/inspected to get `/ostree` ?

Reply:
Are you using this at the same time as https://github.com/osbuild/bootc-image-builder/pull/138 ?

I think we need both right?

I'm always up for a quick realtime sync/debugging session on things like this btw!

Reply:
@cgwalters Yes, this stage is only useful once https://github.com/osbuild/bootc-image-builder/pull/138 has landed.

Reply:
FTR we had a quick realtime sync on this and the conclusion I think was that this approach is fine for now - we can expect that bib is going to guarantee that the source and root here are the same.

However, we also determined that `org.osbuild.container-deploy` today is copying a `podman mount` into the target, and this is discarding the information about what the source container image metadata (specifically the manifest digest).  We could do something like this:

```
diff --git a/stages/org.osbuild.container-deploy b/stages/org.osbuild.container-deploy
index 466c9b9b..08f3456e 100755
--- a/stages/org.osbuild.container-deploy
+++ b/stages/org.osbuild.container-deploy
@@ -72,6 +72,7 @@ def main(inputs, output):
             )
         with mount_container(image_tag) as img:
             subprocess.run(["cp", "-a", f"{img}/.", f"{output}/"], check=True)
+        # write an xattr to the root with the manifest digest
 
 
 if __name__ == "__main__":
```

And then we could perhaps teach the bubblewrap code to detect that the xattr exists and synthesize a `/run/containerenv` file that looks like what podman does.

However, I had not realized until now that osbuild was running `podman mount` - if we're involving podman here, then in practice why aren't we just doing `podman run <container>` to start?  What is the value in doing the complex dance of:

- fetch to `dir` via skopeo
- copy to `containers-storage`
- mount
- copy to flattened temporary tree

versus just:

- (optionally) fetch to external cache directory via skopeo
- if external cache exists, use it, otherwise `podman run`

?

Reply:
And just for the record - the code works with these bind mounts because osbuild/bib sets up everything in exactly the right way :) My comment was more about additional guardrails (that are not strictly needed)

Reply:
The other thing I guess here to guarantee that the source and target are the same would be to just bind mount in the whole root, and not have separate copies at all right?

But when we do that, it still leads me to wonder why we aren't using `podman run` instead of bwrap.

Reply:
`podman run` instead of `bwrap` is a fundamental change to `osbuild`. I strongly believe we simply don't have time to design this and implement it properly within the tight deadlines.

`podman run` in `bwrap` was something that @achilleas-k tried and he had some serious issues with it, so we just scraped it, and went for this compromise.

Yes, we can definitely optimize the copies later.

Reply:
Forgot to add that `podman mount` interestingly works just fine inside `bwrap`.

Reply:
There is a lot of good content in there, my immediate concern about "/ostree" and "--image-sourceref" is resolved so I will close the discussion. There is more insightful comments here, I will get back to that but it all followup work it seems.